src/Tools/dotnet-format/MSBuild/MSBuildEnvironment.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+// Original License:
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+// See https://github.com/dotnet/docfx/blob/d4b1e3015d7c527057193974042ba657da96656c/src/Microsoft.DocAsCode.Metadata.ManagedReference/MsBuildEnvironmentScope.cs
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Text.RegularExpressions;
+using RunTests;
+
+namespace Microsoft.CodeAnalysis.Tools.MSBuild
+{
+    internal class MSBuildEnvironment
+    {
+        private static readonly Regex DotnetBasePathRegex = new Regex("Base Path:(.*)$", RegexOptions.Compiled | RegexOptions.Multiline);
+        private static string DotnetBasePath;
+
+        /// <summary>
+        /// Ensures the proper MSBuild environment variables are populated.
+        /// </summary>
+        public static void ApplyEnvironmentVariables()
+        {
+            var environment = GetEnvironmentVariables();
+            foreach (var kvp in environment)
+            {
+                Environment.SetEnvironmentVariable(kvp.Key, kvp.Value);
+            }
+        }
+
+        private static ImmutableDictionary<string, string> GetEnvironmentVariables()
+        {
+            // workaround for https://github.com/dotnet/docfx/issues/1752
+            var dotnetBasePath = GetDotnetBasePath();
+            if (dotnetBasePath != null)
+            {
+                return new Dictionary<string, string>
+                {
+                    ["MSBUILD_EXE_PATH"] = dotnetBasePath + "MSBuild.dll",
+                    ["MSBuildExtensionsPath"] = dotnetBasePath,
+                    ["MSBuildSDKsPath"] = dotnetBasePath + "Sdks"
+                }.ToImmutableDictionary();
+            }
+
+            return ImmutableDictionary<string, string>.Empty;
+        }
+
+        public static string GetDotnetBasePath()
+        {
+            if (string.IsNullOrEmpty(DotnetBasePath))
+            {
+                DotnetBasePath = DetermineDotnetBasePath();
+            }
+
+            return DotnetBasePath;
+        }
+
+        private static string DetermineDotnetBasePath()
+        {
+            var processResult = ProcessRunner.CreateProcess("dotnet", "--info", captureOutput: true, displayWindow: false).Result.Result;
```

Comment:
Does this launch with the user's project path as the CWD so that it gets the correct version of the .NET CLI per any global.json they might have in their path?

Reply:
Thanks for reviewing. I updated things to pass the workspace's directory in as the working directory.